### PR TITLE
Ensure mscorlib is included when retrying expression evaluation and ignoring unreferenced assemblies

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @KevinH-MS, @ManishJayaswal, @weshaggard, @davkean

**Comment by cston** [comment]
@tmat

File: src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs [hunk 1]
```diff
@@ -95,8 +96,14 @@ internal static class MetadataUtilities
                 }
                 if (identitiesBuilder != null)
                 {
-                    var identity = metadata.MetadataReader.ReadAssemblyIdentityOrThrow();
+                    var reader = metadata.MetadataReader;
+                    var identity = reader.ReadAssemblyIdentityOrThrow();
                     identitiesBuilder.Add(identity);
+                    // If this assembly has no references, assume it is mscorlib.
+                    if (reader.AssemblyReferences.Count == 0)
+                    {
```
Comment (davkean) [inline src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs]: I don't understand where or how this is run, but be aware that's valid to have an assembly without any references, and you do see this in the wild with satellites.

**Comment by KevinH-MS** [comment]
LGTM :shipit:

File: src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs (hunk 1 above)
Comment (tmat) [inline src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs]: The compiler also checks if the assembly defines System.Object, we should do that here too.
See PEAssembly.DeclaresTheObjectClass

Comment (cston) [inline src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs]: Added check. Thanks.

**Pull request merged**